src/main/java/solution/BacktrackingSolution.java

```diff
@@ -0,0 +1,42 @@
+package solution;
+
+import java.awt.*;
```

Comment:
Do you use the https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/master/ide-configuration/intellij-configuration/code-style/intellij-code-style_droolsjbpm-java-conventions.xml? The "star import" (give me everything from this package) is considered a bad practice. It makes it difficult to understand the origin of classes being used in your code.

The second concern is that you actually use it to import `Point`. Although `java.awt` is still part of the standard library, here is doesn't fit much, as `awt` is a library for creating GUI. As you just need a class holding two indexes, why not creating one? It's not so much code to worry about maintainability.